zerver/webhooks/bitbucket2/view.py

```diff
@@ -110,7 +113,6 @@ def get_subject_based_on_type(payload: Dict[str, Any], type: str) -> Text:
     return get_subject(payload)
 
 def get_type(request: HttpRequest, payload: Dict[str, Any]) -> str:
-    event_key = request.META.get("HTTP_X_EVENT_KEY")
```

Comment:
Can you explain why moving this further down is the right thing to do?  

I feel like this is a better thing to validate first, both to avoid duplicated code, and also because this is sorta an initial sanity check that the event is intended for the bitbucket webhook...

Reply:
@timabbott: Well, moving it further down makes sense here because in the bitbucket2 webhook, we don't always rely on the header to figure out the event type, so it makes sense to only get the header when needed. I made a comment about this further down! :) We might want to change that somewhere along the way, but that should be its own PR!

Reply:
Are there payloads where they don't send the event type header at all?  Regardless of how we do our parsing, I think it makes sense to fail here with a clear exception about the header being missing, rather than failing further on because of one of the other checks to determine event type.

Reply:
Okay, so there aren't any payloads that don't send the event type header. It's just that we choose not to rely on the header for most part. I have no idea why that is, the person who wrote this integration just chose to figure out the event type from the payload JSON itself and not from the header (which would have been easier)! The way I see it, we have two options:

* Keep the status quo and simply raise an exception if the header is missing, like you suggested. This would mean passing in the correct headers in the tests, which isn't the case right now, since the header isn't relied upon to determine the even type.
* The other option is to rewrite `get_type` from scratch and make sure that we use the event header to determine the event type. This is a larger project we should eventually get to, but isn't super high priority.

What do you think? :)

Reply:
I think option 1 is fine for now; it's kinda the first step regardless.